<launch>
    <arg name="robot" default="march4" doc="The robot to run. Can be: march3, march4, test_joint_linear, test_joint_rotational."/>
    <arg name="robot_description" default="$(arg robot)" doc="URDF file to use"/>
    <arg name="controller_name" default="$(arg robot)" doc="Name of the controller file"/>
    <arg name="reset_imc" default="false" doc="Reset the IMC if this argument is set to true"/>
    <arg name="if_name" default="" doc="Ethernet port to use, can be found by running 'ifconfig' in the terminal.
                                        If not provided, then the value is read from the robot .yaml file."/>

    <rosparam file="$(find march_hardware_interface)/config/$(arg controller_name)/controllers.yaml" command="load"/>

    <param name="robot_description" textfile="$(find march_description)/urdf/$(arg robot_description).urdf"/>

    <group ns="march">
        <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

        <node name="controller_spawner" pkg="controller_manager" type="controller_manager" respawn="false"
              output="screen"
              args="spawn controller/joint_state controller/temperature_sensor controller/trajectory"
        />

        <node
                launch-prefix="ethercat_grant"
                name="hardware_interface"
                pkg="march_hardware_interface"
                type="march_hardware_interface_node"
                args="$(arg robot)"
                output="screen"
                required="true"
        >
            <param name="reset_imc" value="$(arg reset_imc)"/>
            <param name="if_name" value="$(arg if_name)"/>
        </node>
    </group>
</launch>
